nimble_builder/__init__.py

```diff
@@ -1,3 +1,153 @@
+"""
+On loading nimble_builder the RackParameters dataclass will be available
+"""
+from dataclasses import dataclass
+from typing import Literal
 
-beam_width = 20
-corner_fillet = 2
+@dataclass
+class RackParameters:
```

Comment:
Will this data be passed to ExSource/cq-cli? If it will be, should this have something like a `json` property that returns all the normal properties and derived properties as a JSON string? I don't think derived properties will be picked up in the JSON conversion of a class, correct?

Reply:
(Frustratingly I just wrote a long comment and tried to use it to resolve the conversation and the terrible GitHub interface killed my comment)
My thought is that this will handle all parameters, both fixed ones that will never change and ones that will change, and derived parameters, so that any parameter needed for multiple different components gets its information from a single source of truth.
A sub-set of the parameters will be variable. Exactly how we split and very these, I don't yet know. When we know we should clarify the doc-string.
Rather than add to this specific PR, I have added a task to #58